runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -410,14 +410,28 @@ local function on_line_impl(self, win, buf, line, on_spell, on_conceal)
 
             local spell, spell_pri_offset = get_spell(capture_name)
 
+            local is_noconceal = capture_name == 'noconceal'
+            -- The "conceal" attribute can be set at the pattern level or on a particular capture
+            local conceal_attr = metadata.conceal ~= nil and metadata.conceal
+              or metadata[capture] and metadata[capture].conceal
```

Comment:
this logic looks suspicious. what is the purpose of the `metadata.conceal ~= nil` check, since if `metadata.conceal=false` it will fallthrough to the latter part anyway? also, use parens for clarity, since `a and b or c and d` is not obvious.

Reply:
yea that's it, i can also just create an if statement so that it's abundantly clear:

```lua
local conceal_attr = metadata.conceal
if conceal_attr == nil then
  conceal_attr = metadata[capture] and metadata[capture].conceal
end
```

Reply:
not asking for an if statement, just parens